src/graphql/entries.ts

```diff
@@ -30,51 +40,56 @@ export async function updateEntry({
   updateFromEntryEditor,
   locale,
   entityReferenceMap,
+  gqlParams,
 }: UpdateEntryProps): Promise<Entity & { sys: SysProps }> {
   if (dataFromPreviewApp.sys.id !== updateFromEntryEditor.sys.id) {
     return dataFromPreviewApp;
   }
+  const copyOfDataFromPreviewApp = clone(dataFromPreviewApp);
+  const typename = generateTypeName(contentType.sys.id);
 
-  const copyOfDataFromPreviewApp = { ...dataFromPreviewApp };
-  const { fields } = contentType;
-
-  logUnrecognizedFields(
-    fields.map((f) => f.apiName ?? f.name),
-    dataFromPreviewApp
-  );
-
-  for (const field of fields) {
+  for (const field of contentType.fields) {
     const name = field.apiName ?? field.name;
 
-    if (isPrimitiveField(field)) {
-      updatePrimitiveField(copyOfDataFromPreviewApp, updateFromEntryEditor, name, locale);
-    } else if (field.type === 'RichText') {
-      updateRichTextField({
-        dataFromPreviewApp: copyOfDataFromPreviewApp,
-        updateFromEntryEditor,
-        name,
-        locale,
-      });
-    } else if (field.type === 'Link') {
-      await updateSingleRefField({
-        dataFromPreviewApp: copyOfDataFromPreviewApp,
-        updateFromEntryEditor,
-        name,
-        locale,
-        entityReferenceMap,
-      });
-    } else if (field.type === 'Array' && field.items?.type === 'Link') {
-      await updateMultiRefField({
-        dataFromPreviewApp: copyOfDataFromPreviewApp,
-        updateFromEntryEditor,
-        name,
-        locale,
-        entityReferenceMap,
-      });
+    if (isRequestedField(name, typename, copyOfDataFromPreviewApp, gqlParams)) {
```

Comment:
With a query, we're only resolving requested information - should increase performance